### PR TITLE
docs(component): improves the documentation

### LINKED ISSUE #2481: Clarify whether processors should accept data or not after shutdown
**Is your feature request related to a problem? Please describe.**
Currently the documentation on components don't clarify whether the component should/shouldn't continue accepting data after the Shutdown has been called. This is specially misleading on those cases you want to control the component lifecycle but also want to make sure zero data loss.

**Describe the solution you'd like**
Component documentation makes it clear that it should not accept more data after the shutdown method has been called.

Ping @bogdandrutu

### PR DESCRIPTION
**Pull request by jcchavezs**
Clarifies that components should not accept anymore data on shutdown 

Ping @bogdandrutu 

Closes #2481

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]
Ping @tigrannajaryan

File: component/component.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ type Component interface {
 	// Create a new context from the context.Background() for long-running operations.
 	Start(ctx context.Context, host Host) error
 
-	// Shutdown is invoked during service shutdown.
+	// Shutdown is invoked during service shutdown. After Shutdown() is called, if the component accept data in
```
Comment (jpkrohling) [inline component/component.go:40]: `Shutdown signals the component to stop accepting data and to clean up the resources in use, as the collector is being stopped.`

Comment (tigrannajaryan) [reply component/component.go:40]: If we want to clarify this we also need to tell that any internally stored pdata must be drained, propagated further to the pipeline to the next consumer (or exported out of the Collector if the component is exporter) before Shutdown returns. 
"Clean up the resources" hints to that but I think it is not sufficient.

Comment (jcchavezs) [reply component/component.go:40]: @jpkrohling I would remove the "as the collector is being stopped" that sounds to me too much information from upstream. How about saying "as the processor is being decommissioned"?

@tigrannajaryan this is outlined in https://github.com/open-telemetry/opentelemetry-collector/pull/2504/files#diff-6f065ffd54f98081ffd548c200264f761067d883b72b315c01ac28ef5a8fce47R45-R46 but I like the "before Shutdown returns" part. I think I should add it.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**